File: Web/web-angular/src/app/dto/InitialData.ts
import { GrammarInitial } from './GrammarInitial';

export class InitialData {
    public authors: string[] = [];
    public volumes: string[] = [];
    public statKorpus: StatLine[] = [];
    public statOther: StatLine[] = [];
    public styleGenresParts: string[] = [];
    public styleGenres: { [key:string]:string[]; } = {};
    public grammar: GrammarInitial = new GrammarInitial();
}

export class StatLine {
    public name: string;
    public texts: number;
    public words: number;
}

File: Web/src/org/alex73/korpus/server/data/InitialData.java
package org.alex73.korpus.server.data;

import java.util.List;
import java.util.Map;

import org.alex73.korpus.base.DBTagsGroups.KeyValue;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

@JsonInclude(Include.NON_NULL)
public class InitialData {
    public List<String> authors;
    public List<String> sources;
    public List<KeyValue> subcorpuses;
    public List<String> styleGenresParts;
    public Map<String,List<String>> styleGenres;
    public GrammarInitial grammar;
}
